### PR TITLE
feat(Code): add `isHighlighted` prop

### PR DESCRIPTION
**Pull request by saurabhdaware**
**Changes** (Have discussed these changes with design on figma)
- Adding `isHighlighted` prop to `Code` component which is true by default and can be disabled.
- Default color of `isHighlighted: true` will be `subtle` and default when `isHighlighted: false` will be `normal`
- `color` can be overriden when `isHighlighted` is `false` but not when `isHighlighted` is true
- BackgroundColor when isHighlighted is `true` is now `a100` instead of `a50`

Figma: https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?type=design&node-id=11723-147043&t=C8DJ6KdKlpLotgcV-0

image

### CODE REVIEW

File: packages/blade/src/components/Typography/Code/Code.tsx [hunk 1]
```diff
@@ -63,6 +96,24 @@ const CodeContainer = styled(BaseBox)<CodeContainerProps>((props) => {
   };
 });
 
+const getCodeColor = ({
+  isHighlighted,
+  color,
+}: Pick<CodeProps, 'isHighlighted' | 'color'>): CodeProps['color'] => {
+  if (isHighlighted) {
+    if (color) {
+      throw new Error('[Blade: Code]: `color` prop cannot be used without `isHighlighted={false}`');
```
Comment (anuraghazra) [inline packages/blade/src/components/Typography/Code/Code.tsx:105]: This error sounds negated, can we simplify to: 

```suggestion
      throw new Error('[Blade: Code]: `color` prop cannot be used when `isHighlighted` is true');
```

File: packages/blade/src/components/Typography/Code/Code.tsx [hunk 2]
```diff
@@ -82,33 +133,38 @@ const CodeContainer = styled(BaseBox)<CodeContainerProps>((props) => {
...
   size = 'small',
   weight = 'regular',
-  color = 'surface.text.subtle.lowContrast',
+  isHighlighted = true,
+  color,
   testID,
   ...styledProps
 }: CodeProps): JSX.Element => {
   const { fontSize, lineHeight } = getCodeFontSizeAndLineHeight(size);
+  const codeTextColor: CodeProps['color'] = React.useMemo(
```
Comment (anuraghazra) [inline packages/blade/src/components/Typography/Code/Code.tsx:153]:
```suggestion
  const codeTextColor = React.useMemo<CodeProps['color'] >(
```

File: packages/blade/src/components/Typography/Code/__tests__/Code.native.test.tsx [hunk 3]
```diff
@@ -11,10 +11,26 @@ describe('<Code />', () => {
...
+      <Code isHighlighted={false} color="action.text.link.disabled">
+        TEST_TOKEN
+      </Code>,
+    );
     expect(toJSON()).toMatchSnapshot();
   });
 
+  it('should throw error when color is set without isHighlighted false', () => {
+    const tempConsoleError = console.error;
+    console.error = jest.fn();
```
Comment (anuraghazra) [inline packages/blade/src/components/Typography/Code/__tests__/Code.native.test.tsx:24]: Let's do proper mock? 

```
    const mockConsoleError = jest.spyOn(console, 'error').mockImplementation();
    
    ...
    
    mockConsoleError.mockRestore();

```

File: packages/blade/src/components/Typography/Code/Code.tsx [hunk 4]
```diff
@@ -27,11 +27,42 @@ export type CodeProps = {
...
   size?: 'small' | 'medium';
   weight?: 'regular' | 'bold';
+  isHighlighted?: boolean;
+  color?: BaseTextProps['color'];
 } & TestID &
   StyledPropsBlade;
 
+type CodeHighlightedProps = CodeCommonProps & {
+  /**
+   * Adds code background to highlight the text
```
Comment (anuraghazra) [inline packages/blade/src/components/Typography/Code/Code.tsx:37]: ? 

```suggestion
   * Adds background color to highlight the text
```

**Review by anuraghazra — APPROVED** [review]
Mostly looks fine, minor comments

File: packages/blade/src/components/Typography/Code/Code.tsx (hunk 1 above)
Comment (saurabhdaware) [reply packages/blade/src/components/Typography/Code/Code.tsx:105]: so `isHighlighted` is true by default so people might think "arey?? I didn't make `isHighlighted` true then why is this error coming"

Comment (anuraghazra) [reply packages/blade/src/components/Typography/Code/Code.tsx:105]: Ohh i see, okay.

**Review by divyanshu013 — APPROVED** [review]
Looks good to me.

File: packages/blade/src/components/Typography/Code/Code.tsx [hunk 5]
```diff
@@ -82,33 +133,38 @@ const CodeContainer = styled(BaseBox)<CodeContainerProps>((props) => {
...
   size = 'small',
   weight = 'regular',
-  color = 'surface.text.subtle.lowContrast',
+  isHighlighted = true,
+  color,
   testID,
   ...styledProps
 }: CodeProps): JSX.Element => {
   const { fontSize, lineHeight } = getCodeFontSizeAndLineHeight(size);
+  const codeTextColor = React.useMemo<CodeProps['color']>(
```
Comment (kamleshchandnani) [inline packages/blade/src/components/Typography/Code/Code.tsx:153]: why do we use `useMemo` here? any perf things we witnessed? 🤔

Comment (kamleshchandnani) [reply packages/blade/src/components/Typography/Code/Code.tsx:153]: @saurabhdaware sorry asking on a merged PR but checking all the things that got merged last week 🙈

**Pull request merged**